protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -1494,6 +1494,19 @@ impl ProxyExtendedChannelFactory {
             .clone()
             .ok_or(Error::ShareDoNotMatchAnyJob)?
             .0;
+
+        if referenced_job.job_id != m.job_id {
+            let error = SubmitSharesError {
+                channel_id: m.channel_id,
+                sequence_number: m.sequence_number,
+                error_code: SubmitSharesError::invalid_job_id_error_code()
+                    .to_string()
+                    .try_into()
+                    .unwrap(),
```

Comment:
add a comment that exaplain why the unwrap is safe

Reply:
I agree.

Going forward, IMO we should try to keep some discipline around the usage of `unwrap`, because we are getting some criticism from the community. 

for example: https://github.com/stratum-mining/stratum/issues/723#issuecomment-1904233521